client/cherami/connection.go

```diff
@@ -290,7 +316,25 @@ func (conn *connection) readAcksPump() {
 					default:
 						conn.logger.WithField(common.TagReconfigureID, common.FmtReconfigureID(reconfigInfo.GetUpdateUUID())).Warn("Reconfigure channel is full.  Drop reconfigure command.")
 					}
+				} else if cmd.GetType() == cherami.InputHostCommandType_DRAIN {
+					// drain all inflight messages
+					// reconfigure to pick up new extents if any
+					conn.reporter.IncCounter(metrics.PublishDrainRate, nil, 1)
+					// start draining by just stopping the write pump.
+					// this makes sure, we don't send any new messages.
+					// the read pump will exit when the server completes the drain
+					go conn.stopWritePump()
+
+					reconfigInfo := cmd.Reconfigure
```

Comment:
how about moving this logic into its own method so that it can be shared across the two if branches ?